### PR TITLE
Store visibility status of sidebar menus

### LINKED ISSUE #51: Store visibility status of sidebar menus in settings object
For example:
1. Hide user list.
2. Restart the client.
3. User list is now visible, but if the visibility state was saved, it would stay hidden.

### PR DESCRIPTION
**Pull request by xPaw**
Fixes #51
Closes #1606

Also fixes sidebar reopening on desktop when switching channels with keyboard binds.

### CODE REVIEW

File: client/js/socket-events/init.js [hunk 1]
```diff
@@ -42,12 +42,19 @@ socket.on("init", function(data) {
 
 		webpush.configurePushNotifications(data.pushSubscription, data.applicationServerKey);
 
+		slideoutMenu.enable();
+
+		const viewport = $("#viewport");
+
+		if (viewport.outerWidth() >= 768) {
+			slideoutMenu.toggle(storage.get("thelounge.state.sidebar") === "true");
```
Comment (McInkay) [inline client/js/socket-events/init.js]: Do we need the `=== true`? Surely it'll work fine without?

Comment (xPaw) [reply client/js/socket-events/init.js]: Note the string. false as a string still evaluates to true.

Comment (McInkay) [reply client/js/socket-events/init.js]: Oh stupid.

The purest in me feels like saying use Json.parse then ...

**Comment by astorije** [comment]
Sweet! I noticed a couple bugs though:

- Toggling the user list actually changes the state of the channel list:

  sidebar-bug

- On mobile resolution (I tried both mobile-like in Chrome, and on actual phone), the state is not kept on page refresh. This applies to both channel list and user list, but I only recorded user list:

  sidebar-bug2

**Comment by xPaw** [comment]
> Toggling the user list actually changes the state of the channel list

Brilliant. I'll take a look. This also affects master.

> On mobile resolution (I tried both mobile-like in Chrome, and on actual phone), the state is not kept on page refresh.

This is by design (Anything under 768px is not stored) as I didn't want to store overlays needlessly.

**Comment by astorije** [comment]
> This is by design (Anything under 768px is not stored) as I didn't want to store overlays needlessly.

Aaah, totally missed that. That's fair, and a good starting point.

**Pull request merged**